contracts/voting/Voting.sol

```diff
@@ -0,0 +1,342 @@
+        return _calculateWinner(season, cityId);
+    }
+
+    function calculateVotesPrice(
+        uint256 mayorId,
+        uint256 cityId,
+        uint256 votes
+    ) external view override returns(uint256) {
+        return _calculateVotesPrice(mayorId, cityId, votes);
+    }
+
+    function calculatePrize(uint256 cityId, uint256 season) external view override returns(uint256) {
+        if (!_isRewardPeriod(_cities[cityId].regionId)) revert VotingErrors.IncorrectPeriod();
+        if (mayor.ownerOf(_calculateWinner(season, cityId)) == msg.sender) {
+            uint256 prize = _calculatePrizeToUser(
+                cityId, msg.sender, season, _getBank(_cityToNominees[cityId][season])
+            );
+            return !_ownerClaims[msg.sender][cityId][season] ? prize : 0;
+        }
+    }
+
+    function _startVoting(uint256 regionId) internal {
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentTimestamp = block.timestamp;
+        if (_regions[regionId].startVotingTimestamp == 0) {
+            _regions[regionId] = Region({active: true, startVotingTimestamp: currentTimestamp});
+            emit VotingStarted(regionId, currentTimestamp);
+        }
+    }
+
+    function _seasonNumber(uint256 regionId) internal view returns(uint256) {
+        // solhint-disable-next-line not-rely-on-time
+        return (block.timestamp - _regions[regionId].startVotingTimestamp) /
+            (VOTING_DURATION + GOVERNANCE_DURATION + CLAIMING_DURATION);
+    }
+
+    function _isVotingPeriod(uint256 regionId) internal view returns(bool ok) {
+        // solhint-disable-next-line not-rely-on-time
+        ok = ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+            (VOTING_DURATION + GOVERNANCE_DURATION + CLAIMING_DURATION)) < VOTING_DURATION;
+    }
+
+    function _isGoverningPeriod(uint256 regionId) internal view returns(bool) {
+        /* solhint-disable not-rely-on-time */
+        return 
+            ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+                (VOTING_DURATION + GOVERNANCE_DURATION + CLAIMING_DURATION)) > VOTING_DURATION &&
+            ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+                (VOTING_DURATION + GOVERNANCE_DURATION + CLAIMING_DURATION)) < (VOTING_DURATION + GOVERNANCE_DURATION);
+        /* solhint-enable not-rely-on-time */
+    }
+
+    function _isRewardPeriod(uint256 regionId) internal view returns(bool) {
+        // solhint-disable-next-line not-rely-on-time
+        return ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+            (VOTING_DURATION + GOVERNANCE_DURATION + CLAIMING_DURATION)) > (VOTING_DURATION + GOVERNANCE_DURATION);
+    }
+
+    function _calculateVotesPrice(
+        uint256 mayorId,
+        uint256 cityId,
+        uint256 votes
+    ) internal view returns(uint256) {
+        if (!_isVotingPeriod(_cities[cityId].regionId)) revert VotingErrors.IncorrectPeriod();
```

Comment:
better move this verification out of this function to nominate function. and don't copy it to calculateVotesPrice external function, so it will be callable in any period for information

Reply:
Done